src/main/java/org/alfresco/repo/notification/EMailNotificationProvider.java

```diff
@@ -209,7 +210,17 @@ public void sendNotification(NotificationContext notificationContext)
         {
             mail.setParameterValue(MailActionExecuter.PARAM_SUBJECT, subject);
         }
-        
+
+        // Note that the smart logic in MailActionExecuter should recognize any
+        // preferred locale for the users in the recipient field and override this value
+        // Note that the smart logic in MailActionExecuter may be broken if we send email addresses
+        // instead of userIDs
+        Locale userLocale = I18NUtil.getLocale();

```

Comment:
Are you sure that the correct locale will be set in this thread?

Reply:
Yes, the browser preferences should be available in this thread that handles the request.